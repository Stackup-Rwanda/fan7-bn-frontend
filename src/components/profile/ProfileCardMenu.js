import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './styles/profileCardMenu.scss';

class ProfileCardMenu extends Component {
  constructor() {
    super();
    this.state = {
      isActive: 'activityFeeds',
    };
    this.toggleActiveClass = this.toggleActiveClass.bind(this);
  }

  toggleActiveClass(event) {
    const { handleNavigation } = this.props;
    event.persist();
    this.setState(() => ({
      isActive: event.target.name,
    }));
    handleNavigation(event.target.name);
  }

  render() {
    const { isActive } = this.state;

    return (
      <>
        <button
          type="button"
          className={isActive === 'activityFeeds' ? 'menu active' : 'menu'}
          name="activityFeeds"
          onClick={this.toggleActiveClass}
        >
          Activity Feeds
        </button>
        <button
          type="button"
          className={isActive === 'trips' ? 'menu active' : 'menu'}
          name="trips"
          onClick={this.toggleActiveClass}
        >
          Trips
        </button>
      </>
    );
  }
}

ProfileCardMenu.propTypes = {
  handleNavigation: PropTypes.func.isRequired,
};

export default ProfileCardMenu;
