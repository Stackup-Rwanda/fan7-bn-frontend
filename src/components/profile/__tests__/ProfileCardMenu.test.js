/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount, shallow } from 'enzyme';
import ProfileCardMenu from '../ProfileCardMenu';

describe('ProfileCardMenu Component Test', () => {
  it('should return true if component exists', () => {
    const wrapper = mount(<ProfileCardMenu />);

    expect(wrapper.find('button[name="activityFeeds"]').exists()).toBe(true);
    expect(wrapper.find('button[name="trips"]').exists()).toBe(true);
    expect(wrapper.state().isActive).toEqual('activityFeeds');
  });

  it('should handle toggle active menu', () => {
    const handleNavigation = jest.fn();
    const event = {
      persist: jest.fn(),
      target: {
        name: 'trips',
      },
    };
    const props = {
      handleNavigation,
    };
    const wrapper = shallow(<ProfileCardMenu {...props} />);

    wrapper.find('button[name="trips"]').simulate('click', event);

    expect(handleNavigation).toHaveBeenCalled();
    expect(wrapper.state().isActive).toEqual('trips');
  });
});

describe('<ProfileCardMenu /> shallow rendering tests', () => {
  it('matches the snapshot', () => {
    const props = {
      updateProfile: jest.fn(),
    };
    const tree = shallow(<ProfileCardMenu {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
