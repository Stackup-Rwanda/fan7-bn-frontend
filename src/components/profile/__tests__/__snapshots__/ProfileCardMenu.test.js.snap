// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileCardMenu /> shallow rendering tests matches the snapshot 1`] = `
<Fragment>
  <button
    className="menu active"
    name="activityFeeds"
    onClick={[Function]}
    type="button"
  >
    Activity Feeds
  </button>
  <button
    className="menu"
    name="trips"
    onClick={[Function]}
    type="button"
  >
    Trips
  </button>
</Fragment>
`;
